### PR TITLE
Web configuration editor

### PR DESCRIPTION
**Pull request by KarlStraussberger**
First chunk. Displays most of current config file.

Fixes: #960

### CODE REVIEW

**Comment by KarlStraussberger** [comment]
Web Interface is controlled by gerbera-config.json. Values are taken from current config.

Next step is to add editing features to web ui.
Also I plan to update ConfigManager::load and move default values and some checking to configMap (as vector<ConfigOption>)

**Comment by KarlStraussberger** [comment]
Refactoring ConfigManager is finished for the moment.
ConfigGenerator might be done some time in the future.

Now it's time for the editing feature in UI.

**Comment by KarlStraussberger** [comment]
Most of the options can already be changed (scalar, arrays and dictionaries). No adding for multiple values, yet.

Remaing topics have to wait until the weekend is over.

Next step might be updating the database and make changed persistent.

**Comment by KarlStraussberger** [comment]
The roundtrip to the database is working now (untested for MySQL)

The remainign part is adding and removing values for lists (like mappings).

File: config/sqlite3.sql [hunk 1]
```diff
@@ -55,6 +55,11 @@ CREATE TABLE "mt_metadata" (
   "property_value" text NOT NULL,
   CONSTRAINT "mt_metadata_idfk1" FOREIGN KEY ("item_id") REFERENCES "mt_cds_object" ("id") ON DELETE CASCADE ON UPDATE CASCADE
 );
+CREATE TABLE "mt_config_value" (
```
Comment (whyman) [inline config/sqlite3.sql:58]: Would be `grb_` nowadays surely ;)

Comment (KarlStraussberger) [reply config/sqlite3.sql:58]: I thought about it already, wanted to keep the naming scheme. But, it's not a bad idea to flag our additions correctly. I'll push an update.

Comment (KarlStraussberger) [reply config/sqlite3.sql]: Just experienced an issue that the server hang when a database command caused an exception. Reason behind: The sqliteThread was still running and the exception could not break out.

**Comment by KarlStraussberger** [comment]
It's almost done! (I do not plan to make transcoding items addable)
Next step is to cleanup and add the option "CFG_DIRECTORY_IMPORT" (similar to AutoscanDirectory but overriding lower nesting level settings)

**Comment by whyman** [comment]
I personally dont think we should allow adding autoscans to the config via this method.

You can add autoscan dirs via the UI already, so I thik it would be confusing.

**Comment by KarlStraussberger** [comment]
> 
> 
> I personally dont think we should allow adding autoscans to the config via this method.
> 
> You can add autoscan dirs via the UI already, so I thik it would be confusing.

Autoscan is working now, the only thing I did not implement is transcoding (you can edit existing but not add or remove anything)

**Comment by KarlStraussberger** [comment]
Next 
- Make directoriy tweaking effective in content manager (and where necessary).
- Add dialog to edit tweaks in filesystem view

**Comment by KarlStraussberger** [comment]
Final Steps:
- Cleanup
- Documentation
- Tests for UI

**Comment by KarlStraussberger** [comment]
Last Task:
- Tests for UI

**Comment by KarlStraussberger** [comment]
@whyman I'm satisfied with it now. I'd leave improved user experience (like dropdown lists, range checking) to new PRs

**Comment by whyman** [comment]
Looks like there are conflicts, if you are happy then feel free to merge it

**Comment by neheb** [comment]
```
 error: ‘CFG_SERVER_STORAGE_MYSQL_ENABLED’ was not declared in this scope
         mysql_en = setOption(root, CFG_SERVER_STORAGE_MYSQL_ENABLED)->getBoolOption();
```

**Comment by KarlStraussberger** [comment]
> 
> 
> ```
>  error: ‘CFG_SERVER_STORAGE_MYSQL_ENABLED’ was not declared in this scope
>          mysql_en = setOption(root, CFG_SERVER_STORAGE_MYSQL_ENABLED)->getBoolOption();
> ```

Obviously, I did not compile with MYSQL disabled, it's config.h, line 84. Just swap with line 85.

**Pull request merged**